idaes/core/util/tables.py

```diff
@@ -245,10 +252,14 @@ def create_stream_table_dataframe(
             for i in disp_dict[k]:
                 if i is None:
```

Comment:
from here to line 264 seems like duplicated code except for the difference in the key. why not do that up front and use the same logic, i.e. `units_key = k if is None else f"{k} {i}"` then use 'units_key' in place of 'k' and not require the `if i is None` decision